File: BookShop.Presentation/wwwroot/ClientApp/src/app/models/userModels/passwordChangeRequestModel.ts
export class PasswordChangeRequestModel {
    public currentPassword: string;
    public newPassword: string;
    public newPasswordConfirmation: string;

    constructor() {
        this.initializeDefaultModel();
    }

    private initializeDefaultModel() {
        this.currentPassword = '';
        this.newPassword = '';
        this.newPasswordConfirmation = '';
    }
}

File: BookShop.BusinessLogic/Models/UserModels/PasswordChangeRequestModel.cs
using System.ComponentModel.DataAnnotations;

namespace BookShop.BusinessLogic.Models.UserModels
{
    public class PasswordChangeRequestModel
    {
        [Required]
        public string CurrentPassword { get; set; }
        [Required]
        public string NewPassword { get; set; }
        [Required]
        [Compare("NewPassword")]
        public string NewPasswordConfirmation { get; set; }
    }
}
